extensions/typescript-language-features/src/languageFeatures/diagnostics.ts

```diff
@@ -148,6 +151,87 @@ class DiagnosticSettings {
 	}
 }
 
+class DiagnosticsTelemetryManager extends Disposable {
+
+	private readonly _diagnosticCodesMap = new Map<number, number>();
+	private readonly _diagnosticTimeoutsMap = new Map<URI, NodeJS.Timeout | undefined>();
+	private readonly _diagnosticSnapshotsMap = new Map<URI, readonly vscode.Diagnostic[]>();
+
+	constructor(
+		private readonly _telemetryReporter: TelemetryReporter,
+		private readonly _getDiagnostics: (uri: URI) => readonly vscode.Diagnostic[]
+	) {
+		super();
+		this._register(vscode.workspace.onDidChangeTextDocument(e => {
+			if (e.document.languageId === 'typescript') {
+				this._updateDiagnosticCodesAfterTimeout(e.document.uri, 10000);
+			}
+		}));
+		this._register(vscode.workspace.onDidOpenTextDocument(e => {
+			if (e.languageId === 'typescript') {
+				this._updateDiagnosticCodesAfterTimeout(e.uri, 10000);
+			}
+		}));
+		this._register(vscode.workspace.onDidCloseTextDocument(e => {
+			if (e.languageId === 'typescript') {
+				this._diagnosticTimeoutsMap.delete(e.uri);
+			}
+		}));
+		const activeUri = vscode.window.activeTextEditor?.document.uri;
+		this._updateDiagnosticCodesAfterTimeout(activeUri, 10000);
+		this._sendTelemetryEvent();
+	}
+
+	private _updateDiagnosticCodesAfterTimeout(uri: URI | undefined, timeoutInMs: number) {
```

Comment:
It looks like timeoutInMs is always set to the value 10000. I would suggest to remove that argument